galley/pkg/mcp/client/client.go

```diff
@@ -97,6 +98,51 @@ type Client struct {
 	state      map[string]*perTypeState
 	clientInfo *mcp.Client
 	updater    Updater
+
+	journal  recentRequestsJournal
+	metadata map[string]string
+}
+
+// RecentRequestInfo is metadata about a request that the client has sent.
+type RecentRequestInfo struct {
+	Time    time.Time
+	Request *mcp.MeshConfigRequest
+}
+
+// Acked indicates whether the message was an ack or not.
+func (r RecentRequestInfo) Acked() bool {
+	return r.Request.ErrorDetail != nil
+}
+
+// recentRequestsJournal captures debug metadata about the latest requests that was sent by this client.
+type recentRequestsJournal struct {
+	itemsMutex sync.Mutex
+	items      []RecentRequestInfo
```

Comment:
We can dimension by type, but I left it upto the client. The only thing it affects is, this can cause starvation if one of the types is more active than the rest.

We can plumb that in as a separate check-in, if this becomes a problem.